rbd/pool_nautilus.go

```diff
@@ -46,3 +47,20 @@ func GetPoolMetadata(ioctx *rados.IOContext, key string) (string, error) {
 	}
 	return C.GoString((*C.char)(unsafe.Pointer(&buf[0]))), nil
 }
+
+// SetPoolMetadata updates the pool metadata string associated with the given key.
+//
+// Implements:
+//  int rbd_pool_metadata_set(rados_ioctx_t io_ctx, const char *key, const char *value);
+func SetPoolMetadata(ioctx *rados.IOContext, key, value string) error {
+	cKey := C.CString(key)
+	defer C.free(unsafe.Pointer(cKey))
+	cValue := C.CString(value)
+	defer C.free(unsafe.Pointer(cValue))
+
+	ret := C.rbd_pool_metadata_set(cephIoctx(ioctx), cKey, cValue)
+	if ret < 0 {
+		return RBDError(ret)
```

Comment:
Is there are reason you're not using an error conversion function like `getError`?

Reply:
I thought that getError has more advantage if RBD image is in question which is not the case with these functions.

Reply:
Sorry, I got your point. Ignore the previous comment, will change.